salt/cloud/clouds/clc.py

```diff
@@ -0,0 +1,368 @@
+# -*- coding: utf-8 -*-
+'''
+CenturyLink Cloud Module
+===================
+
+.. versionadded:: 2017.7.14
+
+The CLC cloud module allows you to manage CLC Via the CLC SDK.
+
+:codeauthor: Stephan Looney <slooney@stephanlooney.com>
+
+
+Dependencies
+============
+
+- clc-sdk Python Module
+
+CLC SDK
+-------
+
+clc-sdk can be installed via pip:
+
+.. code-block:: bash
+
+    pip install clc-sdk
+
+.. note::
+  For sdk reference see: https://github.com/CenturyLinkCloud/clc-python-sdk
+
+Configuration
+=============
+
+To use this module, set up the clc-sdk, username,password,key in the
+cloud configuration at
+``/etc/salt/cloud.providers`` or ``/etc/salt/cloud.providers.d/clc.conf``:
+
+.. code-block:: yaml
+
+    my-clc-config:
+      driver: clc
+      user: 'web-user'
+      password: 'verybadpass'
+      token: ''
+      token_pass:''
+
+.. note::
+
+    The ``provider`` parameter in cloud provider configuration was renamed to ``driver``.
+    This change was made to avoid confusion with the ``provider`` parameter that is
+    used in cloud profile configuration. Cloud provider configuration now uses ``driver``
+    to refer to the salt-cloud driver that provides the underlying functionality to
+    connect to a cloud provider, while cloud profile configuration continues to use
+    ``provider`` to refer to the cloud provider configuration that you define.
+
+'''
+
+# Import python libs
+from __future__ import absolute_import
+from random import randint
+from re import findall
+import pprint
+import logging
+import time
+import os.path
+import subprocess
+import json
+# Import salt libs
+import salt.utils
+import salt.utils.cloud
+import salt.utils.xmlutil
+#import pip
+import importlib
+from salt.exceptions import SaltCloudSystemExit
+from flask import Flask, request
+# Import salt cloud libs
+import salt.config as config
+import requests
+# Attempt to import clc-sdk lib
+try:
+        importlib.import_module('clc')
```

Comment:
I'm not clear on why this line is necessary. Could you please clarify?

Reply:
Thanks for the clarification. This is super-weird, though. Is an upstream bug filed with the `clc` maintainers?

Reply:
I submitted a ticket to them via our enterprise account....no updates on it yet.